src/bosh-director/spec/unit/custom_middleware_collector_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'spec_helper'
+
+describe Bosh::Director::CustomMiddlewareCollector do
```

Comment:
The `CustomMiddlewareCollector` class name is very vague and nondescript. Would it be reasonably possible for you to change the name of this class to one that describes its purpose and function?